python312Packages.robotframework: 7.1.1 -> 7.2

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for python312Packages.robotframework is: Generic test automation framework

meta.homepage for python312Packages.robotframework is: https://robotframework.org/

meta.changelog for python312Packages.robotframework is: https://github.com/robotframework/robotframework/blob/master/doc/releasenotes/rf-7.2.rst


###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes





###### Impact

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 7.2 with grep in /nix/store/zajzhww8m2p2g8i10fqv1fri758m573x-python3.12-robotframework-7.2
- found 7.2 in filename of file in /nix/store/zajzhww8m2p2g8i10fqv1fri758m573x-python3.12-robotframework-7.2

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
27 total rebuild path(s)

27 package rebuild(s)

First fifty rebuilds by attrpath

geogram
python312Packages.robomachine
python312Packages.robotframework
python312Packages.robotframework-databaselibrary
python312Packages.robotframework-excellib
python312Packages.robotframework-pythonlibcore
python312Packages.robotframework-requests
python312Packages.robotframework-selenium2library
python312Packages.robotframework-seleniumlibrary
python312Packages.robotframework-sshlibrary
python312Packages.robotstatuschecker
python312Packages.robotsuite
python313Packages.robomachine
python313Packages.robotframework
python313Packages.robotframework-databaselibrary
python313Packages.robotframework-excellib
python313Packages.robotframework-pythonlibcore
python313Packages.robotframework-requests
python313Packages.robotframework-selenium2library
python313Packages.robotframework-seleniumlibrary
python313Packages.robotframework-sshlibrary
python313Packages.robotstatuschecker
python313Packages.robotsuite
renode
renode-unstable
robotframework-tidy
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from the cache**:
```
nix-store -r /nix/store/zajzhww8m2p2g8i10fqv1fri758m573x-python3.12-robotframework-7.2 \
  --option binary-caches 'https://cache.nixos.org/ https://nixpkgs-update-cache.nix-community.org/' \
  --option trusted-public-keys '
  nixpkgs-update-cache.nix-community.org-1:U8d6wiQecHUPJFSqHN9GSSmNkmdiFW7GW7WNAnHW0SM=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The nixpkgs-update cache is only trusted for this store-path realization.)
For the cached download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A python312Packages.robotframework https://github.com/r-ryantm/nixpkgs/archive/ef6361bf439cc8b225f71546356fd8fd2a3a41ef.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/ef6361bf439cc8b225f71546356fd8fd2a3a41ef#python312Packages.robotframework
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/zajzhww8m2p2g8i10fqv1fri758m573x-python3.12-robotframework-7.2
ls -la /nix/store/zajzhww8m2p2g8i10fqv1fri758m573x-python3.12-robotframework-7.2/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:x: 12 packages failed to build:</summary>
  <ul>
    <li>python312Packages.robotframework-selenium2library</li>
    <li>python312Packages.robotframework-selenium2library.dist</li>
    <li>python312Packages.robotframework-seleniumlibrary</li>
    <li>python312Packages.robotframework-seleniumlibrary.dist</li>
    <li>python313Packages.robotframework-pythonlibcore</li>
    <li>python313Packages.robotframework-pythonlibcore.dist</li>
    <li>python313Packages.robotframework-selenium2library</li>
    <li>python313Packages.robotframework-selenium2library.dist</li>
    <li>python313Packages.robotframework-seleniumlibrary</li>
    <li>python313Packages.robotframework-seleniumlibrary.dist</li>
    <li>robotframework-tidy</li>
    <li>robotframework-tidy.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 41 packages built:</summary>
  <ul>
    <li>geogram</li>
    <li>geogram.bin</li>
    <li>geogram.dev</li>
    <li>geogram.doc</li>
    <li>geogram.lib</li>
    <li>python312Packages.robomachine</li>
    <li>python312Packages.robomachine.dist</li>
    <li>python312Packages.robotframework</li>
    <li>python312Packages.robotframework-databaselibrary</li>
    <li>python312Packages.robotframework-databaselibrary.dist</li>
    <li>python312Packages.robotframework-excellib</li>
    <li>python312Packages.robotframework-excellib.dist</li>
    <li>python312Packages.robotframework-pythonlibcore</li>
    <li>python312Packages.robotframework-pythonlibcore.dist</li>
    <li>python312Packages.robotframework-requests</li>
    <li>python312Packages.robotframework-requests.dist</li>
    <li>python312Packages.robotframework-sshlibrary</li>
    <li>python312Packages.robotframework-sshlibrary.dist</li>
    <li>python312Packages.robotframework.dist</li>
    <li>python312Packages.robotstatuschecker</li>
    <li>python312Packages.robotstatuschecker.dist</li>
    <li>python312Packages.robotsuite</li>
    <li>python312Packages.robotsuite.dist</li>
    <li>python313Packages.robomachine</li>
    <li>python313Packages.robomachine.dist</li>
    <li>python313Packages.robotframework</li>
    <li>python313Packages.robotframework-databaselibrary</li>
    <li>python313Packages.robotframework-databaselibrary.dist</li>
    <li>python313Packages.robotframework-excellib</li>
    <li>python313Packages.robotframework-excellib.dist</li>
    <li>python313Packages.robotframework-requests</li>
    <li>python313Packages.robotframework-requests.dist</li>
    <li>python313Packages.robotframework-sshlibrary</li>
    <li>python313Packages.robotframework-sshlibrary.dist</li>
    <li>python313Packages.robotframework.dist</li>
    <li>python313Packages.robotstatuschecker</li>
    <li>python313Packages.robotstatuschecker.dist</li>
    <li>python313Packages.robotsuite</li>
    <li>python313Packages.robotsuite.dist</li>
    <li>renode</li>
    <li>renode-unstable</li>
  </ul>
</details>

---

###### Maintainer pings

cc @bjornfor for [testing](https://github.com/ryantm/nixpkgs-update/blob/main/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

> [!TIP]
> As a maintainer, if your package is located under `pkgs/by-name/*`, you can comment **`@NixOS/nixpkgs-merge-bot merge`** to automatically merge this update using the [`nixpkgs-merge-bot`](https://github.com/NixOS/nixpkgs-merge-bot).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

r-ryantm committed:
python312Packages.robotframework: 7.1.1 -> 7.2

bjornfor reviewed (changes requested):
Out of the 12 build errors, two of them are regressions:

* robotframework-tidy
* robotframework-tidy.dist

They start failing with this error:

```
Checking runtime dependencies for robotframework_tidy-4.15.0-py3-none-any.whl                                                                                                                                            
   - robotframework<7.2,>=4.0 not satisfied by version 7.2                                                                                                                                                                
```

NickCao committed:
robotframework-tidy: 4.15.0 -> 4.16.0

Diff: https://github.com/MarketSquare/robotframework-tidy/compare/refs/tags/4.15.0...4.16.0

Changelog: https://github.com/MarketSquare/robotframework-tidy/blob/main/docs/releasenotes/4.16.0.rst

NickCao committed:
python312Packages.robotframework: modernize

bjornfor reviewed (commented):
@NickCao: Thanks!

Can you put the robotframework-tidy update first, to avoid regressions between the commits?

NickCao commented:
> Can you put the robotframework-tidy update first, to avoid regressions between the commits?

Sure.

bjornfor commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 375022`

---
### `x86_64-linux`
<details>
  <summary>:x: 10 packages failed to build:</summary>
  <ul>
    <li>python312Packages.robotframework-selenium2library</li>
    <li>python312Packages.robotframework-selenium2library.dist</li>
    <li>python312Packages.robotframework-seleniumlibrary</li>
    <li>python312Packages.robotframework-seleniumlibrary.dist</li>
    <li>python313Packages.robotframework-pythonlibcore</li>
    <li>python313Packages.robotframework-pythonlibcore.dist</li>
    <li>python313Packages.robotframework-selenium2library</li>
    <li>python313Packages.robotframework-selenium2library.dist</li>
    <li>python313Packages.robotframework-seleniumlibrary</li>
    <li>python313Packages.robotframework-seleniumlibrary.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 43 packages built:</summary>
  <ul>
    <li>geogram</li>
    <li>geogram.bin</li>
    <li>geogram.dev</li>
    <li>geogram.doc</li>
    <li>geogram.lib</li>
    <li>python312Packages.robomachine</li>
    <li>python312Packages.robomachine.dist</li>
    <li>python312Packages.robotframework</li>
    <li>python312Packages.robotframework-databaselibrary</li>
    <li>python312Packages.robotframework-databaselibrary.dist</li>
    <li>python312Packages.robotframework-excellib</li>
    <li>python312Packages.robotframework-excellib.dist</li>
    <li>python312Packages.robotframework-pythonlibcore</li>
    <li>python312Packages.robotframework-pythonlibcore.dist</li>
    <li>python312Packages.robotframework-requests</li>
    <li>python312Packages.robotframework-requests.dist</li>
    <li>python312Packages.robotframework-sshlibrary</li>
    <li>python312Packages.robotframework-sshlibrary.dist</li>
    <li>python312Packages.robotframework.dist</li>
    <li>python312Packages.robotstatuschecker</li>
    <li>python312Packages.robotstatuschecker.dist</li>
    <li>python312Packages.robotsuite</li>
    <li>python312Packages.robotsuite.dist</li>
    <li>python313Packages.robomachine</li>
    <li>python313Packages.robomachine.dist</li>
    <li>python313Packages.robotframework</li>
    <li>python313Packages.robotframework-databaselibrary</li>
    <li>python313Packages.robotframework-databaselibrary.dist</li>
    <li>python313Packages.robotframework-excellib</li>
    <li>python313Packages.robotframework-excellib.dist</li>
    <li>python313Packages.robotframework-requests</li>
    <li>python313Packages.robotframework-requests.dist</li>
    <li>python313Packages.robotframework-sshlibrary</li>
    <li>python313Packages.robotframework-sshlibrary.dist</li>
    <li>python313Packages.robotframework.dist</li>
    <li>python313Packages.robotstatuschecker</li>
    <li>python313Packages.robotstatuschecker.dist</li>
    <li>python313Packages.robotsuite</li>
    <li>python313Packages.robotsuite.dist</li>
    <li>renode</li>
    <li>renode-unstable</li>
    <li>robotframework-tidy</li>
    <li>robotframework-tidy.dist</li>
  </ul>
</details>

bjornfor reviewed (approved):
The remaining build errors are not regressions.
